Use the `VMediaCollection` for search and collection results

obulat opened the pull request:
<!-- prettier-ignore -->
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, please consider opening one before creating this pull request. -->
Based on #3831
Fixes #3884 by @obulat
Fixes #3576 by @AetherUnbound
Fixes #3830 by @obulat

_Initially, I split the changes into the search result and collection result changes. However, this made it more difficult to understand the scope and changes, and to test the PRs. This is why I merged the collection changes (as well as the deletion of the now-unused components) into this PR._

## Description
<!-- Concisely describe what the pull request does. -->
<!-- Add screenshots, videos, or other media to show the problem and the solution when appropriate. -->
This PR fixes the collection result fetching and improves the fetching of the search results. 


It uses the new `VMediaCollection` component and 2 new components for the search and collection results.
- `VSearchResults` component fills in the components necessary for search results: search title and content links in the header, load more button, and the external search form in the footer.
- `VCollectionResults` component adds the components necessary for collection results: `VCollectionHeader` and load more button.

The media fetching was updated: now it is done in a single place: `search.vue` and `collection.vue`, instead of being split among these pages, middleware and the Load more component. This allows us to handle fetching, refetching, and loading more items from the same place. The "Load more" button now only handles sending the Analytics events.

## Testing Instructions
<!-- Give steps for the reviewer to verify that this PR fixes the problem; or delete this section entirely. -->
Run the app with `additional_search_views` on.
Check that the search results are updated correctly: when the search term is updated, search type is changed, filters are updated or load more button is clicked. The result labels should be updated correctly, loading skeletons shown when media is fetching for the first time, media should be updated for the correct query.
The same check needs to be done for collection pages

The search results should be updated correctly when the `additional_search_views` is off.

Repeat the instructions from #3884 and see that the problem is fixed and results are refreshed

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. -->

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

zackkrida reviewed (changes requested):
Before I dive too deeply into the code I am seeing two bugs. Let me know if this PR is the right place to address these; if not, I am happy to ignore them and re-review!

1. There is a regression where the placeholder "skeleton" UI isn't shown when switching media types via the content switcher.
2. When navigating to a source collection from a creator collection, the results are not re-fetched.

obulat commented:
>     1. There is a regression where the placeholder "skeleton" UI isn't shown when switching media types via the content switcher.

This should be addressed in this PR

>     2. When navigating to a source collection from a creator collection, the results are not re-fetched.

Should be fixed in #3886

I'm going to draft this PR to address the first point.

obulat commented:
>     1. There is a regression where the placeholder "skeleton" UI isn't shown when switching media types via the content switcher.

It turns out there were too many `mediaStore.clearMedia` calls, which removed the existing media in the store while also resetting the fetch state. So, `fetchSingleMedia` would set `isFetching` to `true`, and then it would immediately be reset to `false` by another call (when switching type or in the `search.vue` route watcher). 
I audited the `clearMedia` calls and removed unnecessary ones.

>     2. When navigating to a source collection from a creator collection, the results are not re-fetched.
When I first started implementing this PR, we had no "source" link button on the creator page, so there was no way of navigating from one collection to another. You could only navigate from a collection to a single result. That's why I didn't even test this scenario.
To fix, I added a route watcher similar to the one in the `search.vue` component that fetches the media if the route changes.

obulat commented on frontend/src/stores/media/index.ts:
```diff
@@ -451,6 +456,7 @@ export const useMediaStore = defineStore("media", {
     }) {
       const searchStore = useSearchStore()
       const queryParams = searchStore.getApiRequestQuery(mediaType)
+      console.log("fetchSingleMediaType", mediaType, queryParams)
```
```suggestion
```

zackkrida commented on frontend/src/stores/search.ts:
```diff
@@ -149,6 +149,11 @@ export const useSearchStore = defineStore("search", {
     state.recentSearches = useStorage<string[]>("recent-searches", [])
   },
   getters: {
+    searchStarted(state) {
+      return state.strategy === "default"
+        ? state.searchTerm.length > 0
+        : state.collectionParams !== null
+    },
```
Nit: I feel like a comment could be useful to explain what this behavior is for. So basically `searchStarted` is a boolean used in calculating `canLoadMore`? and `searchStarted` represents that on standard views, there is a search term, and on collection pages, that the required params are present? Just making sure I follow :smile:

obulat replied:
I removed this in favor of `fetchState.hasStarted` in the media store, as the purpose of that property is to see if the fetching was started.

zackkrida reviewed (approved):
This LGTM with the caveat that there's a typo in the translation paths which I have outlined [here](https://github.com/WordPress/openverse/pull/3835/files#r1528731896). I will pre-approve though since that's only a typo. Everything is behaving well.

obulat commented on frontend/src/pages/image/collection.vue:
```diff
@@ -1,27 +1,100 @@
+      }
+      const { collection, ...params } = collectionParams.value
+      return i18n
+        .t(`collection.label.${collection}.image`, { ...params })
```
Thank you so much for providing the correct key! :)

dhruvkb reviewed (approved):
LGTM!
